### PR TITLE
batmon: init at 0.0.1

### PR DESCRIPTION
**Pull request by 6543**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ba/batmon/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage rec {
```
Comment (fabaff) [inline pkgs/by-name/ba/batmon/package.nix:6]:
```suggestion
}:

rustPlatform.buildRustPackage rec {
```

File: pkgs/by-name/ba/batmon/package.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  pname = "batmon";
+  version = "0.0.1";
+
+  src = fetchFromGitHub {
+    owner = "6543";
+    repo = pname;
+    rev = "v${version}";
```
Comment (fabaff) [inline pkgs/by-name/ba/batmon/package.nix:13]:
```suggestion
    repo = "batmon";
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/ba/batmon/package.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    hash = "sha256-+kjDNQKlaoI5fQ5FqYF6IPCKeE92WKxIhVCKafqfE0o=";
+  };
+  cargoSha256 = "sha256-DJpWBset6SW7Ahg60+Tu1VpH34LcVOyrEs9suKyTE9g=";
```
Comment (fabaff) [inline pkgs/by-name/ba/batmon/package.nix:16]:
```suggestion
  };

  cargoSha256 = "sha256-DJpWBset6SW7Ahg60+Tu1VpH34LcVOyrEs9suKyTE9g=";
```

File: pkgs/by-name/ba/batmon/package.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description = "An interactive viewer, similar to top, htop and other *top utilities, but about the batteries installed in your notebook.";
```
Comment (fabaff) [inline pkgs/by-name/ba/batmon/package.nix:19]:
```suggestion
    description = " Interactive batteries viewer ";
    longDescription = ''
      An interactive viewer, similar to top, htop and other *top utilities,
      but about the batteries installed in your notebook.
    '';
```

File: pkgs/by-name/ba/batmon/package.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = "https://crates.io/crates/batmon";
```
Comment (fabaff) [inline pkgs/by-name/ba/batmon/package.nix:20]:
```suggestion
    homepage = "https://github.com/6543/batmon/";
    changelog = "https://github.com/6543/batmon/releases/tag/v${version}";
```

File: pkgs/by-name/ba/batmon/package.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    license = licenses.asl20;
+    mainProgram = "batmon";
+    maintainers = with maintainers; [_6543];
```
Comment (fabaff) [inline pkgs/by-name/ba/batmon/package.nix:23]:
```suggestion
    maintainers = with maintainers; [ _6543 ];
```

**Comment by 6543** [comment]
the tool itselve should work on darwin too ... but I dont have any mac to test the nix pkg build etc ...

File: pkgs/by-name/ba/batmon/package.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    description = "Interactive batteries viewer";
+    longDescription = ''
+      An interactive viewer, similar to top, htop and other *top utilities,
+      but about the batteries installed in your notebook.
+    '';
+    homepage = "https://github.com/6543/batmon/";
+    changelog = "https://github.com/6543/batmon/releases/tag/v${version}";
+    license = licenses.asl20;
+    mainProgram = "batmon";
+    platforms = platforms.linux;
```
Comment (h7x4) [inline pkgs/by-name/ba/batmon/package.nix:29]: If it should be usable on darwin, you should probably modify this to include it.

Comment (6543) [reply pkgs/by-name/ba/batmon/package.nix:29]: I dont have a way to test it, feel free to push to it @h7x4

Comment (h7x4) [reply pkgs/by-name/ba/batmon/package.nix:29]: I am not able to push to it, as the `Allow edits from maintainers` box seems to be unchecked. But I had a look upstream, and found out that `starship-battery` supports a whole range of systems.

```suggestion
    platforms = with platforms; unix ++ windows;
```

Comment (6543) [reply pkgs/by-name/ba/batmon/package.nix:29]: Yes but the darwin builder (x86) failed and some told me it does not work on the darwin-arm arch ...

Comment (6543) [reply pkgs/by-name/ba/batmon/package.nix:29]: I can apply but we will see the builder fail again ... :/

Comment (h7x4) [reply pkgs/by-name/ba/batmon/package.nix]: If `aarch64-darwin` is broken, you should set `meta.broken = stdenv.isDarwin && stdenv.isAarch64;`. `meta.platforms` is usually meant to tell you what platforms the program is meant for and supports upstream, not what we are able to build.

**Comment by 6543** [comment]
-> new pkg

**Pull request merged**